internal/network/manager.go

```diff
@@ -19,189 +20,172 @@ import (
 	m = &Manager{
 		iceNotifier:              ntf,
 		bufferTransports:         make(map[uint32]chan<- *rtp.Packet),
 		bufferTransportGenerator: btg,
 		dataChannelEventHandler:  dcet,
 	}
-	m.dtlsState, err = dtls.NewState(m.handleDTLSState)
-	if err != nil {
-		return nil, err
-	}
 
 	m.sctpAssociation = sctp.NewAssocation(m.dataChannelOutboundHandler, m.dataChannelInboundHandler, m.handleSCTPState)
 
-	m.IceAgent = ice.NewAgent(m.iceNotifier)
-	for _, i := range localInterfaces() {
-		p, portErr := newPort(i+":0", m)
-		if portErr != nil {
-			return nil, portErr
-		}
-
-		m.ports = append(m.ports, p)
-		m.IceAgent.AddLocalCandidate(&ice.CandidateHost{
-			CandidateBase: ice.CandidateBase{
-				Protocol: ice.ProtoTypeUDP,
-				Address:  p.listeningAddr.IP.String(),
-				Port:     p.listeningAddr.Port,
-				Conn:     p.conn,
-			},
-		})
-	}
+	m.IceAgent = ice.NewAgent(urls, m.iceNotifier)
 
 	return m, err
 }
 
-func (m *Manager) handleDTLSState(state dtls.ConnectionState) {
-	if state == dtls.Established {
-		m.sctpAssociation.Connect()
-	}
-}
-
 func (m *Manager) handleSCTPState(state sctp.AssociationState) {
 	if state == sctp.Established {
 		// Temporary way to signal sending OpenChannel messages
 		m.dataChannelEventHandler(&DataChannelOpen{})
 	}
 }
 
-// AddURL takes an ICE Url, allocates any state and adds the candidate
-func (m *Manager) AddURL(url *ice.URL) error {
-	switch url.Scheme {
-	case ice.SchemeTypeSTUN:
-		laddr, xoraddr, err := webrtcStun.AllocateUDP(url)
-		if err != nil {
-			return err
-		}
+// Start allocates DTLS/ICE state that is dependent on if we are offering or answering
+func (m *Manager) Start(isOffer bool, remoteUfrag, remotePwd string, dtlsCert *x509.Certificate, dtlsPrivKey crypto.PrivateKey) error {
+	m.isOffer = isOffer
 
-		p, err := newPort(laddr.String(), m)
-		if err != nil {
-			return err
-		}
+	// Start the sctpAssociation
+	m.sctpAssociation.Start(isOffer)
 
-		c := &ice.CandidateSrflx{
-			CandidateBase: ice.CandidateBase{
-				Protocol: ice.ProtoTypeUDP,
-				Address:  xoraddr.IP.String(),
-				Port:     xoraddr.Port,
-				Conn:     p.conn,
-			},
-			RemoteAddress: laddr.IP.String(),
-			RemotePort:    laddr.Port,
-		}
+	// Spin up ICE
+	var iceConn *ice.Conn
+	var err error
+	if isOffer {
+		iceConn, err = m.IceAgent.Dial(context.TODO(), remoteUfrag, remotePwd)
+	} else {
+		iceConn, err = m.IceAgent.Accept(context.TODO(), remoteUfrag, remotePwd)
+	}
 
-		m.portsLock.Lock()
-		defer m.portsLock.Unlock()
-		m.ports = append(m.ports, p)
-		m.IceAgent.AddLocalCandidate(c)
-	default:
-		return errors.Errorf("%s is not implemented", url.Scheme.String())
+	if err != nil {
+		return err
 	}
 
-	return nil
-}
+	m.iceConn = iceConn
+
+	// Spin up SRTP
+	srtpConn := srtp.Wrap(iceConn, m.handleSRTP)
+
+	// Spin up DTLS
+	var dtlsConn *dtls.Conn
+	if isOffer {
+		// Assumes we offer to be passive and this is accepted.
+		// TODO: Cert?
```

Comment:
I don't understand the `// TODO: Cert?` is this leftover before we finished cert support in `pions/dtls` ?